packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp

```diff
@@ -133,34 +133,36 @@ namespace Tpetra {
     //! Calls doTargetToSource() and then activates the source map mode
     void endFill()
     {
-      if(*activeMultiVector_ == FE_ACTIVE_TARGET) {
-        doTargetToSource(Tpetra::ADD);
+      if(*activeMultiVector_ == FE_ACTIVE_OWNED_PLUS_SHARED) {
+        doOwnedPlusSharedToOwned(Tpetra::ADD);
         switchActiveMultiVector();
       }
       else
-        throw std::runtime_error("FEMultiVector: Source MultiVector already active.  Cannot endFill()");
+        throw std::runtime_error("FEMultiVector: Owned+Shared MultiVector already active.  Cannot endFill()");
     }
 
     //! Activates the target map mode
     void beginFill()
     {
-      // Note: This does not throw an error since the on construction, the FEMV is in target mode.  Ergo, calling beginFill(),
+      // Note: This does not throw an error since the on construction, the FEMV is in owned+shared  mode.  Ergo, calling beginFill(),
       // like one should expect to do in a rational universe, should not cause an error.
-      if(*activeMultiVector_ == FE_ACTIVE_SOURCE) {
+      if(*activeMultiVector_ == FE_ACTIVE_OWNED) {
         switchActiveMultiVector();
       }
     }
 
+    /// \brief Migrate data from the owned+shared to the owned multivector
+    /// Since this is non-unique -> unique, we need a combine mode.
+    /// Precondition: Must be FE_ACTIVE_OWNED_PLUS_SHARED mode
+    /// \warning This is for EXPERT USE ONLY.  We make NO PROMISES of
+    ///   backwards compatibility.
+    void doOwnedPlusSharedToOwned(const CombineMode CM=Tpetra::ADD);
```

Comment:
Here is where a different CombineMode could actually make sense.  (I don't think it makes sense for CrsGraph.)